src/appSelector.ts

```diff
@@ -518,6 +606,102 @@ function getAppVariants(appdir: string, appName: string, folderUri: vscode.Uri):
   return variants;
 }
 
+// Get pytest tests list
+export function getAppTestsList(targetSelector: TargetSelector, showMenu: boolean = false) {
+  let testsList: string[] = [];
+  if (
+    selectedApp
+    && selectedApp.functionalTestsDir
+    && selectedApp.containerName
+    && targetSelector.getSelectedTarget() !== "All"
+  ) {
+    let lastTests = getSetting("testsList", selectedApp.folderUri) as string[];
+    let lastSelectedTests = getSetting("selectedTests", selectedApp.folderUri) as string[];
+    vscode.commands.executeCommand("setContext", "ledgerDevTools.showSelectTests", false);
+    selectedApp.functionalTestsList = [];
+    selectedApp.selectedTests = [];
+    let device = targetSelector.getSelectedSpeculosModel();
+    let optionsExec: cp.ExecOptions = { cwd: selectedApp!.folderUri.fsPath, windowsHide: true };
+    // If platform is windows, set shell to powershell for cp exec.
+    if (platform === "win32") {
+      let shell: string = "C:\\windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe";
+      optionsExec.shell = shell;
+    }
+    // Find the tests list with pytest :
+    // * Get the device option from pytest help (either --model or --device)
+    // * If the device option is found, run pytest with --collect-only and the device option
+    // * If the device option is not found, run pytest with --collect-only
+    let getTestsListCmd = "docker";
+
+    const varPrefix = process.platform === "win32" ? "\`$" : "$";
+    const quotesAroundBashCommand = process.platform === "win32" ? "\"" : "";
+    let getTestsListArgs = [
+      "exec", "-u", "0", selectedApp!.containerName, "bash", "-c",
```

Comment:
Do we really need to start in `root`?

Reply:
Yes to install the requirements with `pip install -r requirements.txt`, if we don't install the requirements, tests collection with `pytest --collect-only` might fail (as pytest needs to execute the tests files to collect the tests, even if not executing the tests themselves)